src/coreclr/vm/typehandle.cpp

```diff
@@ -343,6 +344,93 @@ BOOL TypeHandle::IsCanonicalSubtype() const
     return (*this == TypeHandle(g_pCanonMethodTableClass)) || IsSharedByGenericInstantiations();
 }
 
+#ifndef DACCESS_COMPILE
+bool TypeHandle::IsManagedClassObjectPinned() const
+{
+    LIMITED_METHOD_DAC_CONTRACT;
+
+    // Function pointers are always mapped to typeof(IntPtr)
+    return !GetLoaderAllocator()->CanUnload() || IsFnPtrType();
+}
+
+void TypeHandle::AllocateManagedClassObject(LoaderAllocator* allocator, RUNTIMETYPEHANDLE* pDest, TypeHandle type)
```

Comment:
You do not need to pass loader allocator as separate argument. It is easy to get it from TypeHandle inside the implementation.

Reply:
Made it an instance method of TypeHandle (or it's too much?) 🙂